### ISSUE TITLE
"loss_coord" does not going down

### ISSUE THREAD

**Issue by Secondgrade** [SangbumChoi/MobileHumanPose#17]
Hi,
Really appreciate your efforts on this repo. I'm following your work to generate a model for my own "toy" project. But I can't reach the same metrics as you did. The loss_coord keeps fluctuating around 0.4-0.5 even though the learning rate decayed to 1e-5.
This is my settings:

    ## input, output
    input_shape = (256, 256) 
    output_shape = (input_shape[0]//8, input_shape[1]//8)
    width_multiplier = 1.0
    depth_dim = 32
    bbox_3d_shape = (2000, 2000, 2000) # depth, height, width
    pixel_mean = (0.485, 0.456, 0.406)
    pixel_std = (0.229, 0.224, 0.225)

    ## training config
    embedding_size = 2048
    lr_dec_epoch = [17, 21]
    end_epoch = 25
    lr = 1e-3
    lr_dec_factor = 10
    batch_size = 24

Thanks

**Comment by SangbumChoi** [comment]
@Secondgrade 
Hi thanks for using my repo. There are two things that want to check and share.
1. My repo is about to refactored with minor bugs (if exist)
2. loss can fluctuate even though lr decrease. It would be more plausible to share in a viewpoint of MPJPE

**Comment by Secondgrade** [comment]
> @Secondgrade Hi thanks for using my repo. There are two things that want to check and share.
> 
> 1. My repo is about to refactored with minor bugs (if exist)
> 2. loss can fluctuate even though lr decrease. It would be more plausible to share in a viewpoint of MPJPE

Thanks your response, the following pic is my experiment from epoch 0 to 49. I trained all 32 joints of Human36M. And I also trained the 17 joints model, the trend of the loss is pretty close to the 32 joints model.

![MPJPE](https://user-images.githubusercontent.com/48266815/144798464-91f0f064-7ef7-48ff-add5-df3afcf6dc27.jpg)

Do you mind share your loss_coord when your model reach the metric of Protocol 2 in paper.?

**Comment by SangbumChoi** [comment]
> > @Secondgrade Hi thanks for using my repo. There are two things that want to check and share.
> > 
> > 1. My repo is about to refactored with minor bugs (if exist)
> > 2. loss can fluctuate even though lr decrease. It would be more plausible to share in a viewpoint of MPJPE
> 
> Thanks your response, the following pic is my experiment ofprotocol 2 from epoch 0 to 49 on all 32 joints of Human36M. And I also trained the 17 joints model, the trend of the loss is pretty close to the 32 joints model and the minimal MPJPE of protocol 2 is 67.43.
> 
> ![MPJPE](https://user-images.githubusercontent.com/48266815/144798464-91f0f064-7ef7-48ff-add5-df3afcf6dc27.jpg)
> 
> Do you mind share your loss_coord when your model reach the metric of Protocol 2 in paper.?

For the case of joint number of 17, if you didn't change the setting of my configuration then it should be around 30~ minival MPJPE for protocol 2. However what I don't understand is what do you mean by 32 joints?

**Comment by SangbumChoi** [comment]
@Secondgrade ![image](https://user-images.githubusercontent.com/34004152/144948584-6dce6d36-8430-47be-a6e1-2b4e4a44c0df.png)

This is the one that is not used in official paper

**Comment by Secondgrade** [comment]
> > > @Secondgrade Hi thanks for using my repo. There are two things that want to check and share.
> > > 
> > > 1. My repo is about to refactored with minor bugs (if exist)
> > > 2. loss can fluctuate even though lr decrease. It would be more plausible to share in a viewpoint of MPJPE
> > 
> > 
> > Thanks your response, the following pic is my experiment ofprotocol 2 from epoch 0 to 49 on all 32 joints of Human36M. And I also trained the 17 joints model, the trend of the loss is pretty close to the 32 joints model and the minimal MPJPE of protocol 2 is 67.43.
> > ![MPJPE](https://user-images.githubusercontent.com/48266815/144798464-91f0f064-7ef7-48ff-add5-df3afcf6dc27.jpg)
> > Do you mind share your loss_coord when your model reach the metric of Protocol 2 in paper.?
> 
> For the case of joint number of 17, if you didn't change the setting of my configuration then it should be around 30~ minival MPJPE for protocol 2. However what I don't understand is what do you mean by 32 joints?

32 joints means that I used all joints that Human36M official annotation files provided. And I tried the 17 joints version annotations, but the MPJPE of protocol 2 is 67.4.

**Comment by SangbumChoi** [comment]
#10 There is a slight issue because of batch size and initialzation problem with different users. However, MPJPE with 67.4 seems that somethings wrong with your setting. 

But there is possibility that I did something wrong with the code while refactoring. If it is true, then it will be refactored ASAP
